tycho-common/src/dto.rs

```diff
@@ -1635,6 +1635,89 @@ impl ComponentTvlRequestResponse {
     }
 }
 
+/// Request body for fetching snapshots of protocol states and contract states.
+///
+/// This combines multiple data sources into a single request:
+/// - Protocol states for the specified components
+/// - Contract states (VM storage) for the specified contracts
+/// - Component TVL (if requested)
+/// - Traced entry points (if chain is Ethereum)
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, ToSchema, Default)]
+pub struct SnapshotRequestBody {
+    #[serde(default)]
+    pub chain: Chain,
+    /// Protocol system name, required for correct state resolution
+    #[serde(alias = "protocolSystem")]
+    pub protocol_system: String,
+    /// Component IDs to fetch protocol states for
+    #[serde(alias = "componentIds", default)]
+    pub component_ids: Vec<String>,
+    /// Contract addresses to fetch VM storage for
+    #[serde(alias = "contractIds", default)]
+    #[schema(value_type=Vec<String>)]
+    pub contract_ids: Vec<Bytes>,
+    /// Block number for versioning
+    #[serde(alias = "blockNumber")]
+    pub block_number: u64,
+    /// Whether to include balance information
+    #[serde(alias = "includeBalances", default)]
+    pub include_balances: bool,
+    /// Whether to fetch TVL data
+    #[serde(alias = "includeTvl", default)]
+    pub include_tvl: bool,
+}
+
+impl SnapshotRequestBody {
+    pub fn new(
+        chain: Chain,
+        protocol_system: String,
+        component_ids: Vec<String>,
+        contract_ids: Vec<Bytes>,
+        block_number: u64,
+        include_balances: bool,
+        include_tvl: bool,
+    ) -> Self {
+        Self {
+            chain,
+            protocol_system,
+            component_ids,
+            contract_ids,
+            block_number,
+            include_balances,
+            include_tvl,
+        }
+    }
+}
+
+/// Response containing snapshot data for protocol states and VM storage.
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, ToSchema)]
+pub struct SnapshotRequestResponse {
```

Comment:
I think these structs should not be within `dto` but more central. The RPC client uses indeed a lot of `dto` structs because it communicates between the indexer via JSON. But in this case it doesn't fit imo. The snapshot retrieval is on higher level and it's **not** necessary to parse this struct from a JSON. Nor is it necessary to encode the `SnapshotRequestBody` from a JSON.

Ideally in the future the client should also use higher level structs as input and as output and convert between those using dto structs...